doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,216 @@
+```
+
+```
+key                      = VNET_ROUTE_TUNNEL_TABLE:vnet_name:prefix ; Vnet route tunnel table with prefix 
+; field                  = value 
+ENDPOINT                 = list of ipv4 addresses    ; comma separated list of endpoints 
+ENDPOINT_MONITOR         = list of ipv4 addresses    ; comma separated list of endpoints 
+MAC_ADDRESS              = 12HEXDIG                  ; Inner dst mac in encapsulated packet 
+VNI                      = DIGITS                    ; VNI value in encapsulated packet 
+WEIGHT                   = DIGITS                    ; Weights for the nexthops, comma separated (Optional) 
+PROFILE                  = STRING                    ; profile name to be applied for this route, for community  
+                                                       string etc (Optional) 
+```
+
+### BFD
+
+```
+BFD_SESSION:{{ifname}}:{{prefix}}  
+    "tx_interval": {{interval}} (OPTIONAL) 
+    "rx_interval": {{interval}} (OPTIONAL)  
+    "multiplier": {{detection multiplier}} (OPTIONAL) 
+    "shutdown": {{false}} 
+    "multihop": {{false}} 
+    "local_addr": {{ipv4/v6}} (OPTIONAL) 
+    "type": {{string}} (active/passive..) 
+; Defines APP DB schema to initiate BFD session.
+```
+
+## 2.3 Module Interaction
+
+Overlay routes can be programmed via RestAPI or gNMI/gRPC interface which is not described in this document. A highlevel module interaction is shown below
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_ModuleInteraction.png)
+
+## 2.3 Orchestration Agent
+Following orchagents shall be modified. 
+
+### VnetOrch
+
+#### Requirements
+
+- Vnetorch to add support to handle multiple endpoints for APP_VNET_RT_TUNNEL_TABLE_NAME based route task. 
+- Reuse Nexthop tunnel based on the endpoint configuration. 
+- If there is already the same endpoint exists, use that as member for Nexthop group. 
+- Similar to above, reuse nexthop group, if multiple routes are programmed with the same set of nexthops. 
+- Provide support for endpoint modification for a route prefix. Require SAI support for SET operation of routes. 
+- Provide support for endpoint deletion for a route prefix. Orchagent shall check the existing entries and delete any tunnel/nexthop based on the new route update
+- Ensure backward compatibility with single endpoint routes
+- Use SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT for specifying weights to nexthop member
+- Desirable to have per tunnel stats via sai_tunnel_stat_t 
+ 
+#### Detailed flow
+
+VnetOrch is one of the critical module for supporting overlay ecmp. VnetOrch subscribes to VNET and ROUTE updates from APP_DB.
+
+When a new route update is processed by the add operation,  
+
+1. VnetOrch checks the nexthop group and if it exists, reuse the group 
+2. For a new nexthop group member, add the ECMP member and identify the corresponding monitoring IP address. Create a mapping between the monitoring IP and nexthop tunnel endpoint. 
```

Comment:
Do we have any dependency with nexthop group member? we just have to start the BFD session for that particular VTEP from HW (offloading) or FRR BFD?

Reply:
Thats right, just have to start the session for that VTEP via HW offload